src/v/kafka/server/tests/produce_consume_utils.cc

```diff
@@ -353,4 +358,44 @@ kafka_consume_transport::raw_consume_from_partition(
     co_return records;
 }
 
+ss::future<kafka::offset> kafka_produce_transport::produce_to_partition(
+  const model::ntp& ntp, model::record_batch batch) {
+    if (ntp.ns != model::kafka_namespace) {
+        throw std::runtime_error(
+          fmt::format("cannot produce to ntp with namespace {}", ntp.ns));
+    }
+    return produce_to_partition(
+      ntp.tp.topic, ntp.tp.partition, std::move(batch));
+}
+
+ss::future<kafka::offset> kafka_consume_transport::timequery(
```

Comment:
ICYMI something like this exists in https://github.com/redpanda-data/redpanda/blob/dev/src/v/kafka/server/tests/list_offsets_utils.cc